cmd/ncproxy/ncproxygrpc/networkconfigproxy.proto

```diff
@@ -0,0 +1,131 @@
+syntax = "proto3";
+
+package ncproxygrpc;
+
+service NetworkConfigProxy {
+    rpc AddNIC(AddNICRequest) returns (AddNICResponse) {}
+    rpc DeleteNIC(DeleteNICRequest) returns (DeleteNICResponse) {}
+    rpc CreateNetwork(CreateNetworkRequest) returns (CreateNetworkResponse) {}
+    rpc CreateEndpoint(CreateEndpointRequest) returns (CreateEndpointResponse) {}
+    rpc AddEndpoint(AddEndpointRequest) returns (AddEndpointResponse) {}
+    rpc DeleteEndpoint(DeleteEndpointRequest) returns (DeleteEndpointResponse) {}
+    rpc DeleteNetwork(DeleteNetworkRequest) returns (DeleteNetworkResponse) {}
+    rpc GetEndpoint(GetEndpointRequest) returns (GetEndpointResponse) {}
+    rpc GetNetwork(GetNetworkRequest) returns (GetNetworkResponse) {}
+    rpc GetEndpoints(GetEndpointsRequest) returns (GetEndpointsResponse) {}
+    rpc GetNetworks(GetNetworksRequest) returns (GetNetworksResponse) {}
+}
+
+message AddNICRequest {
+    string container_id = 1;
+    string nic_id = 2;
+    string endpoint_name = 3;
+}
+
+message AddNICResponse {}
+
+message DeleteNICRequest {
+    string container_id = 1;
+    string nic_id = 2;
+    string endpoint_name = 3;
+}
+
+message DeleteNICResponse {}
+
+message CreateNetworkRequest {
+    enum NetworkMode
+    {
+        Transparent = 0;
+    }
+
+    enum IpamType 
+    {
+        Static = 0;
+        DHCP = 1;
+    }
+
+    string name = 1; 
+    NetworkMode mode = 2;
+    string switch_name = 3;
+    IpamType ipam_type = 4;
+    repeated string subnet_ipadress_prefix = 5;
+    string default_gateway = 6;
+}
+
+message CreateNetworkResponse{
+    string id = 1;
+}
+
+message CreateEndpointRequest {
+    enum EndpointPolicyType
+    {
+        PortName = 0;
+    }
+
+    message PortNameEndpointPolicySetting {
+        string port_name = 1;
+    }
```

Comment:
We should include the other policy types here that are supported (such as IOV :P ) and change `PortnameEndpointPolicySetting portname_policy_setting` to an Any type called `policy_settings`

Reply:
I think we probably only want to support policy types that we need to, that will give us greater flexibility in the future.

Do we expect to only ever configure a single policy at a time through this? If so, can we use a [oneof](https://developers.google.com/protocol-buffers/docs/proto3#oneof) instead? If we need multiple policies at once, do we need the `policy_type` field?

Reply:
@kevpar Agreed, only two so far are iov and portname. This was locked in before the iov changes were really taken into account and all we cared about was portname. I think ideally the ep policies should be provided in a repeated message. However each policy has a different number of configurable settings so not sure how to model the message. I don't want to just take a JSON blob and unmarshal into X based on the endpoint policy type and hope for the best

Reply:
A few options would be:

Multiple fields:

```
message CreateEndpointRequest
{
    [...]
    PortNamePolicy port_name_policy = 1;
    IovPolicy iov_policy = 2;
    FooPolicy foo_policy = 3;
    [...]
}
```

Repeated `oneof`:

```
message Policy
{
    oneof policy
    {
        PortNamePolicy port_name = 1;
        IovPolicy iov = 2;
        FooPolicy foo = 3;
    }
}

message CreateEndpointRequest
{
    [...]
    repeated Policy policies = 1;
    [...]
}
```

Repeated `Any`:

```
message CreateEndpointRequest
{
    [...]
    repeated google.protobuf.Any policies = 1;
    [...]
}
```

Reply:
I guess you might still need a way to make each field in approach 1 optional, but if we can do that, I think it's my preferred approach.

Reply:
I think it will generate as a pointer field for each, so we should be okay.

Reply:
2 seems the most straightforward to me and let's us add policies as we see fit without letting anything to be passed in. Do you have a preferred approach? Otherwise I'll go with 2. Me and Kathryn had planned to add the IOV work right after this got in if that's fine to change in the next PR.

Reply:
Oh I didn't see your other messages when I posted mine

Reply:
2 is weird in that it would allow the same policy type to be passed multiple times. Is there anything special about "policies" that makes us want to express them as something other than just an embedded message directly on the request?

Reply:
I see where you're coming from now, and no not really. I'd like to get Mohans input on this, just messaged him.

Reply:
Mohans in favor of approach 1, I'll send an email in the AM with the proposed change